javascript/javascript2/week3/warmUp.js

```diff
@@ -0,0 +1,192 @@
+// 1. Log out the text Called after 2.5 seconds 2.5 seconds after the script is loaded.
+
+setTimeout(afterSeconds, 2500)
+
+function afterSeconds() {
+    console.log('Called after 2.5 seconds')
+}
+
+// 2. Create a function that takes 2 parameters: delay and stringToLog. Calling this function should log out the
+//    stringToLog after delay seconds. Call the function you have created with some different arguments.
+
+function logOutWithDelay(delay, stringToLog) {
+    function stringToLogFunction() {
+        console.log(stringToLog)
+    }
+    delayWithSeconds = delay * 1000
+    setTimeout(stringToLogFunction, delayWithSeconds)
+}
+
+logOutWithDelay(3, 'This string logged after 3 seconds')
+logOutWithDelay(5, 'This string logged after 5 seconds')
+
+// 3. Create a button in html. When clicking this button, use the function you created in the previous task to log out the text:
+//    Called after 5 seconds 5 seconds after the button is clicked.
+
+const button = document.getElementById('clickButton')
+
+button.addEventListener('click', () => {
+    logOutWithDelay(5, 'This string logged after 5 seconds')
+})
+
+
+// 4. Create two functions and assign them to two different variables. One function logs out Earth,
+//    the other function logs out Saturn. Now create a new third function that has one parameter: planetLogFunction.
+//    The only thing the third function should do is call the provided parameter function. Try call the third function
+//    once with the Earth function and once with the Saturn function.
+
+function earthLogger() {
+    console.log('Earth')
+}
+
+function saturnLogger() {
+    console.log('Saturn')
+}
+
+function planetLogFunction(functionForRun) {
+    const result = functionForRun()
+    return result
+}
```

Comment:
You don't have to return anything here.
And if you do, you can combine the lines:
```js
function planetLogFunction(functionForRun) {
    return functionForRun()
}
```